storage/innobase/dict/dict0load.cc

```diff
@@ -2479,17 +2479,20 @@ static dict_table_t *dict_load_table_one(const span<const char> &name,
 			buf_block_t* block = buf_page_get(
 				page_id, table->space->zip_size(),
 				RW_S_LATCH, &mtr);
-			const bool corrupted = !block
-				|| page_get_space_id(block->page.frame)
-				!= page_id.space()
-				|| page_get_page_no(block->page.frame)
-				!= page_id.page_no()
-				|| (mach_read_from_2(FIL_PAGE_TYPE
-						    + block->page.frame)
-				    != FIL_PAGE_INDEX
-				    && mach_read_from_2(FIL_PAGE_TYPE
-							+ block->page.frame)
-				    != FIL_PAGE_TYPE_INSTANT);
+			bool corrupted = !block;
+			if (block) {
```

Comment:
What was the downside of the old code? It used short-circuiting to check if block was valid before attempting operations on it. Unless there was some out of order execution going on, which I believe should not happen for logic operations, it was fine as it was. The new code that uses an if offers a chance for branch misprediction IMHO.